File: 2551. Put Marbles in Bags.py
#* 2551. Put Marbles in Bags
weights = [1,3,5,1]; k = 2
from typing import List
class Solution:
    def putMarbles(self, weights: List[int], k: int) -> int:
        if (k == 1):
             return 0
        w = [weights[i] + weights[i+1] for i in range(len(weights)-1)]
        w.sort()
        return sum(w[-(k-1):]) - sum(w[:k-1])

print(Solution().putMarbles(weights,k))

File: 2551. Put Marbles in Bags.rb
#* 2551. Put Marbles in Bags
weights = [1,3,5,1]; k = 2
# @param {Integer[]} weights
# @param {Integer} k
# @return {Integer}
def put_marbles(weights, k)
    w = weights.each_cons(2).collect(&:sum).sort
    w.last(k-1).sum - w.first(k-1).sum
end
p put_marbles(weights, k)